SWIPs/swip-multiple_payment_processing_support.md

```diff
@@ -35,41 +35,152 @@ Incorporating the required abstractions to support payment modules will require
-* Sending the recipient a payment.
-* Returning true when the payment was successful.
-* Returning false when the payment was not successful.
-* Verifying any on-chain conditions and making sure that they don't change within a reasonable time window after the transaction happens (due to block reorganizations / payment-module specific conditions)
-* Referencing a type, version and base currency 
+* Sending a payment to the recipient.
+* Handling any potential errors.
+* Verifying any on-chain conditions and making sure that they don't change within a reasonable time window after the transaction happens (due to block reorganizations / payment-module specific conditions).
 * Optionally exposing other methods such as querying balances, topping up balances or sending payments (outside of Swarm). 
 
-Nodes can specify their preference for both payment module, as well as price oracle in a list in a configuration file. The preferences are normalized and weighed.
-For any preference list, the chosen option will be the option which has the highest cumulative preference. The preference list has three dimensions, which will be resolved from high to low:
+There are three dimensions to take into consideration for the payment preferences: 
 
 1. Currency to use
 2. Price oracle to use
 3. Payment method to use
 
-Nodes must be able to reach an agreement in any dimension of the preference list, or the fallback option for each dimension of the preference list will be chosen:
+Nodes should be allowed to specify their payment preferences and indicate which dimension (or dimensions) will have the biggest impact when negotiating the payment mechanism to use with their peers.
+
+The following is an example of the proposed configuration:
+
+```yaml
+# How important a dimension is for the node at the moment of negotiating with a peer the payment mechanism to use
+dimensions:
+	currency: 70
+	provider: 20
+	oracle: 10
+
+# Supported payment options by the node.
+# A payment is defined by a currency, the weight (preference) for such currency and the preferred providers and oracles.
+# Provider preference for a given currency is defined by the order of the provider. As an example, for the rif currency Lumino has a  
+# higher preference over Raiden. 
+payments:
+	rif:
+		weight: 36
+		providers:
+			lumino:
+				weigth: 15
+			raiden:
+				weight: 45
+		oracles:
+			oracleA:
+				weigth: 10
+			oracleB:
+				weigth: 70
+			oracleC:
+				weigth: 60
+	xdai:
+		weight: 9
+		providers:
+			Lumino: 
+				weigth: 10
+		oracles:
+			oracleA: 
+				weight: 20
+	dai:
+		weight: 5
+		providers:
+			raiden:
+				weight: 80
+		oracles:
+			oracleC: 
+				weight: 30
+			oracleE:
+				weight: 20
+	eth:
+		weight: 40
+		providers:
+			swap:
+				weight: 35
+		oracles:
+			oracleB:
+				weight: 90
+			oracleE:
+				weight: 10
+```
+
+It's important to mention that weights can be any numeric value and that they are normalized.
```

Comment:
Even though they can be any numeric value, I would propose to keep the sum of any dimension 100. Thus: making the total sum of payments 100, total sum of providers_rif 100, oracles_rif 100, providers_xdai 100, etc.

This will make the table easier to read while just looking at it. You can still mention obviously that this is not a strict requirement, as it is normalized anyways.